### DISCUSSION TITLE
ObjectStore to share audio/video files between microservies

### DISCUSSION THREAD

**Discussion by ealeykin** [nats-io/nats-server#5262] [Q&A] [answered]
Is it a viable approach in terms of performance, maintenance etc.?

An alternative would be to use minio, s3, nfs, but since we already have nats with object store available - what would you suggest ?

Thanks
Eugene

**Comment by derekcollison** [comment, marked as answer]
We have folks using it for this use case yes. 

Currently the system uses ordered consumers to get the objects so that they are flow controlled end to end. With KV and direct GETS the system through NATS core can find the nearest replicable using DQ etc.

Since ObjectStore uses consumers, if you have many digital twins / mirrors you will need to attach to the closest replica. This is not too bad, when you ask for StreamInfo (ObjectInfo) for the origin ObjectStore, it will list all know alternates, the digital twins / mirrors in order of RTT preference from your request, meaning your app.

You can then bind to that one for low latency access to the objects.
